manifests/init.pp

```diff
@@ -87,4 +91,31 @@
       before  => File['satellite_config_yaml'],
     }
   }
+
+  if $trusted_external_command {
```

Comment:
Would it make sure to ensure the integration files are absent in the `else` case?

Reply:
It's a little tricky. I've updated the parameter name to `$manage_trusted_external_command` (which matches with the other Boolean parameter in the class) which I think should reduce the intuition that there is any de-configuration being offered here. Does that still feel like it needs a de-configuration option?

Part of the trickiness is in management of the surrounding config; namely, the creation of the directory (which should be a Puppet standard thing, not owned by satellite\_pe\_tools), and the management of the puppet.conf setting. I kind of want to avoid the mess of stepping into that in situations where satellite is not the only trusted external command.

Reply:
> Part of the trickiness is in management of the surrounding config; namely, the creation of the directory (which should be a Puppet standard thing, not owned by satellite_pe_tools), and the management of the puppet.conf setting. I kind of want to avoid the mess of stepping into that in situations where satellite is not the only trusted external command.

I get that, which is why I only suggested to ensure the specific command and config file would be absent. Perhaps the directory resource should use `ensure_resource()` to deal with duplication definitions.

Reply:
Hmm. That's an interesting idea. I'm not a fan of `ensure_resource()`, generally speaking. I find that it introduces a Prisoner's Dilemma-like situation to your Puppet code, and is therefore never an ideal approach to a cross-module problem.

I think a longer-term fix which avoids a "kicking the can down the road" feel to it would be to define the dependent configuration in an appropriate upstream module. For PE, that would be in puppetlabs/puppet_enterprise. There is no reason the setting and directory should ever be anything different from the values given here. At that point, we would simply remove the resources from this module.

The only resource that belongs here is `File[/etc/puppetlabs/puppet/trusted-external-commands/satellite]`.